### ISSUE TITLE
1.16.5 Spice of life Carrot edition losing all previously eaten foods therefore losing all health benefits when dying

### ISSUE THREAD

**Issue by Ashley123477** [Teneted/Tenet#1940]
**Minecraft Version :** 1.16.5

**Mohist Version :** 899

**Operating System :** Windows 11

**Concerned mod / plugin** : Spice of life Carrot Edition

**Logs :** None

**Steps to Reproduce :**
 1. Eat enough food to gain a health benefit
 2. Die
 3. Have to eat all the food you have already eaten

**Description of issue :** Eating food gives you extra health once you have eaten enough different foods, but once you die you have to eat all of the foods you've already eaten again, this is a config option but on my server it is set to false so i dont understand why its doing it.

**Comment by djburg** [comment]
Include the error log from console too. Ill post mine if I see it

**Comment by Ashley123477** [comment]
Sadly i wasnt around when the issue happened, so im not sure which log its in.

**Comment by BreizhHardware** [comment]
> Sadly i wasnt around when the issue happened, so im not sure which log its in.

I this is probably not an issue but the mod who is configured like that, in the config file (name sol.toml) you have a lane like reset on death. I hope i have help you.

**Comment by Ashley123477** [comment]
> > Sadly i wasnt around when the issue happened, so im not sure which log its in.
> 
> I this is probably not an issue but the mod who is configured like that, in the config file (name sol.toml) you have a lane like reset on death. I hope i have help you.

Like i said in the original post, this is already set to false.

**Comment by BreizhHardware** [comment]
> > > Sadly i wasnt around when the issue happened, so im not sure which log its in.
> > 
> > I this is probably not an issue but the mod who is configured like that, in the config file (name sol.toml) you have a lane like reset on death. I hope i have help you.
> 
> Like i said in the original post, this is already set to false.

And in the world config (/world/serverconfig/) is correctly configured ?

**Comment by Ashley123477** [comment]
> > > > Sadly i wasnt around when the issue happened, so im not sure which log its in.
> > > 
> > > 
> > > I this is probably not an issue but the mod who is configured like that, in the config file (name sol.toml) you have a lane like reset on death. I hope i have help you.
> > 
> > 
> > Like i said in the original post, this is already set to false.
> 
> And in the world config (/world/serverconfig/) is correctly configured ?

Correct

**Comment by BreizhHardware** [comment]
> > > > > Sadly i wasnt around when the issue happened, so im not sure which log its in.
> > > > 
> > > > 
> > > > I this is probably not an issue but the mod who is configured like that, in the config file (name sol.toml) you have a lane like reset on death. I hope i have help you.
> > > 
> > > 
> > > Like i said in the original post, this is already set to false.
> > 
> > And in the world config (/world/serverconfig/) is correctly configured ?
> 
> Correct

That's really wird, I never have an issue with this mod and Mohist, the last update have probably broke the compatibility

**Comment by Ashley123477** [comment]
i think so

**Comment by MagnusHJensen** [comment]
This ^^ is not a complete fix, nor most likely a proper fix.
However, maybe it can be looked over by someone else, and they could spot a mistake or something.
Anyways, that patch readds the caps after death, but in a non-proper way. And does not handle config options, that block the caps from being re-added in the clone event. (Maybe this was caused by the config not updating when I changed it, so it didn't actually set the proper value)
Nor does it respect if the capability is actually not supposed to be persisted.
